src/fields-match.ts

```diff
@@ -1,21 +1,25 @@
 import { FieldValidationFunction, FieldValidationResult } from 'lc-form-validation';
 
-const isString = (input: any) => typeof input === 'string' || input instanceof String;
 const defaultInvalidMessage = 'Fields do not match';
 
+const isString = (input: any) => typeof input === 'string' || input instanceof String;
+
 export interface Fields {
   first: string;
   second: string;
 }
 
-export const validateFieldsMatch: FieldValidationFunction<Fields> = ({ first, second }) => {
+export const VALIDATION_TYPE = 'FIELDS-MATCH';
+
+export const validateFieldsMatch: FieldValidationFunction = ({ first, second }) => {
```

Comment:
I guess this won't work

FieldValidationFunction has three parameters:

- *value* that has changed
- *vm* viewmodel (all the data from the form, entity)
- *custom params*

It should be something lik

export const validateFieldMatch : FieldValidationFunction(value, vm, customParams);

In custom params you should pass the id of the fields (entity) that need to be compared, in this case we only need the id the of the "repeat email" or "repeat password", becuase the original value we got it already in our value param.

something like

MyViewModel {
  password: string;
  repeatpassword : string;
}

We want repeat password to hold the fields match validation

The call should be something like (let's say we have a viewModel, an input and the fieldId is 'password')

validateField(vm, e.target.value, 'password');

Something we have to consider (maybe is time to open lc-form-validation and add an update), what would happen if we have something like?

```javascript
MyViewModel {
 mysubObject: { 
  password: string;
  },
  repeatpassword : string;
}
``` 
To make this call work

validateField(vm, e.target.value, 'mysubOject.password');

We have to use _lodash_ _getOr_, we have already getOr in lc-form-validation we have to consider wether to add a peer dependency, or expose it in lc-form, something to discuss later on (right now let's cover the basics).

Reply:
@brauliodiez check this when you can, please